Sources/PointFree/Pricing.swift

```diff
@@ -95,22 +95,39 @@ extension Pricing: Codable {
 
 let pricingResponse =
   redirectActiveSubscribers(user: get1)
-    <| writeStatus(.ok)
+    <| fetchCoupon
+    >=> writeStatus(.ok)
     >=> map(lower)
     >>> respond(
       view: pricingView,
-      layoutData: { currentUser, pricing, formFields, route in
+      layoutData: { currentUser, pricing, formStyle, coupon, route in
         SimplePageLayoutData(
           currentRoute: route,
           currentUser: currentUser,
-          data: (currentUser, pricing, formFields, route),
+          data: (currentUser, pricing, formStyle, coupon, route),
           extraStyles: pricingExtraStyles <> whatToExpectStyles,
           style: .base(.minimal(.dark)),
           title: "Subscribe to Point-Free"
         )
     }
 )
 
+private func fetchCoupon<I>(
+  _ conn: Conn<I, Tuple5<Database.User?, Pricing, PricingFormStyle, Stripe.Coupon.Id?, Route?>>
+  ) -> IO<Conn<I, Tuple5<Database.User?, Pricing, PricingFormStyle, Either<String, Stripe.Coupon>?, Route?>>> {
```

Comment:
I know there's probably a nice way of using `over4` and some `sequence`s to compose this, but was struggling a bit and just moved the `over4` inside...

Reply:
should this be a middleware transformers so that we could do the flash when fetching the coupon fails?